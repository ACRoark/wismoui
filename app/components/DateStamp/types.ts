export type DateStampFormat = 'auto' | 'long' | 'short';
