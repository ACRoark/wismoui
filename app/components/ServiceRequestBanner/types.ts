export interface IBannerNotification {
  hasError: boolean;
  message: IMessage;
}

export interface IMessage {
  id: string;
  values: string[] | null;
}
