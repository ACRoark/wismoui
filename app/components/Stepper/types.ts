import { ReactElement } from 'react';
import { Description } from 'types';

export interface IStepDescription {
  description: Description;
  key: string;
  status: StepperStatus;
  title: ReactElement | string;
}

export enum StepperStatus {
  Active = 'active',
  Completed = 'completed',
  Error = 'error',
  Pending = 'pending',
}
