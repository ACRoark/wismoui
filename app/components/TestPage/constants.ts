import IOrderRequest from 'types/IOrderRequest';
import IProduct from '../../types/IProduct';

export const fakeProductList: IProduct[] = [
  {
    productType: 'GCR_A_H',
    serviceDate: '2020-04-22',
  },
  {
    productType: 'EFR_D1',
    serviceDate: '2020-04-22',
  },
  {
    productType: 'EFR_D1_1',
    serviceDate: '2020-04-22',
  },
];

export const fakeServiceRequest: IOrderRequest = {
  accountNumber: '111122233334',
  address: {
    city: 'Bloomfield',
    line1: '123 Main',
    line2: null,
    state: 'MI',
    zip: '48301',
  },
  contactPhoneNumber: '7348675309',
  orderRequestStatusUpdates: [
    {
      createdAtDate: '20200319',
      createdAtTime: '131530',
      orderRequestStatus: 'REQUESTED',
    },
  ],
  orderRequestType: 'MIMO_START',
  premiseId: 'premise123',
  products: [
    {
      productType: 'GCR_A_H',
      serviceDate: '2020-04-22',
    },
    {
      productType: 'EFR_D1',
      serviceDate: '2020-04-22',
    },
    {
      productType: 'EFR_D1_1',
      serviceDate: '2020-04-22',
    },
  ],
  serviceOrderEvents: [
    {
      appointment: {
        date: '2020-04-17',
        slotType: 'AM',
      },
      latestStatus: {
          createdAtDate: '20200305',
          createdAtTime: '223443',
          serviceOrderEventStatus: 'COMPLETED',
        },
        product: 'EFR_D1',
      serviceOrderCategory: null,
      serviceOrderId: '123456789',
    },
  ],
  verificationEvents: [
    {
      verificationCategory: 'R975',
      verificationEventId: '123123',
      verificationEventStatusUpdates: [
        {
          verificationCode: 'AD',
          verificationEventStatus: 'COMPLETED',
          createdAtDate: '20200319',
          createdAtTime: '131530',
        },
      ],
    },
  ],
  wantDate: '2020-04-22',
};

export const fakeStartServiceRequest: IOrderRequest = {
  ...fakeServiceRequest,
  orderRequestType: 'MIMO_START',
};

export const fakeStopServiceRequest: IOrderRequest = {
  ...fakeServiceRequest,
  orderRequestType: 'MIMO_STOP',
};
