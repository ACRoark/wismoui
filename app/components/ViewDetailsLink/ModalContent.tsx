import * as _ from 'lodash';
import React, { FC, ReactElement, useEffect, useState } from 'react';
import { FormattedMessage } from 'react-intl';

import DateStamp from 'components/DateStamp';
import DteContactNumber from 'components/DteContactNumber';
import VerificationHistory from 'components/Timelines/VerificationHistory';
import useConfig from 'hooks/useConfig';
import encryptionApiClient from 'services/encryptionApiClient';
import { IVerificationEvent } from 'types';

import getDteContactNumber from './getDteContactNumber';
import messages from './messages';

interface IModalContentProps {
  event: IVerificationEvent;
  events: IVerificationEvent[];
  orderCreatedDate: string;
}

const ModalContent: FC<IModalContentProps> = (props: IModalContentProps): ReactElement => {
  const { event, events, orderCreatedDate } = props;

  const { urls } = useConfig();
  const [encryptedVerificationEventId, setEncryptedVerificationEventId] = useState('');

  const category = event?.verificationCategory;
  const code = _.last(event?.verificationEventStatusUpdates).verificationCode;
  const contactNumber = getDteContactNumber(category);
  const formattedOrderCreatedDate = orderCreatedDate && <DateStamp key={orderCreatedDate} value={orderCreatedDate} />;
  const showVerificationHistory = !['AP', 'PM', 'WP'].includes(code);
  const verificationEventId = event?.verificationEventId;

  // append the tokenized verification event ID to the url.documentSubmission here
  useEffect((): void => {
    if (verificationEventId) {
      const getEncryptedVerificationEventId = async (): Promise<void> => {
        const encryptedTokenValue = await encryptionApiClient.encryptTokenAsync(
          urls.encryptTokenApi,
          verificationEventId,
        );

        setEncryptedVerificationEventId(encryptedTokenValue);
      };

      getEncryptedVerificationEventId();
    }
  }, []);

  const appendedDocumentSubmissionUrl = `${urls.documentSubmission}${encryptedVerificationEventId}`;

  return (
    <span className="dte-wismo-view-details-modal-content">
      <FormattedMessage
        {...messages[`${code}_${category}_description`]}
        values={{
          a: (...msg: React.ReactNode[]): React.ReactNode => (
            <a
              className="website"
              href={appendedDocumentSubmissionUrl}
              key={`${code}${category}${msg}`}
              target="_blank"
            >
              {msg}
            </a>
          ),
          assistance: (...msg: React.ReactNode[]): React.ReactNode => (
            <a className="website" href={urls.assistanceURL} key={`${code}${category}${msg}`} target="_blank">
              {msg}
            </a>
          ),
          li: (...msg: React.ReactNode[]): React.ReactNode => <li key={`${code}${category}${msg}`}>{msg}</li>,
          login: (...msg: React.ReactNode[]): React.ReactNode => (
            <a className="website" href={urls.signIn} key={`${code}${category}${msg}`} target="_blank">
              {msg}
            </a>
          ),
          p: (...msg: React.ReactNode[]): React.ReactNode => <p key={`${code}${category}${msg}`}>{msg}</p>,
          ol: (...msg: React.ReactNode[]): React.ReactNode => <ol key={`${code}${category}${msg}`}>{msg}</ol>,
          mimo: (...msg: React.ReactNode[]): React.ReactNode => (
            <a className="website" href={urls.backToMIMO} key={`${code}${category}${msg}`} target="_blank">
              {msg}
            </a>
          ),
          orderCreatedDate: formattedOrderCreatedDate,
          ul: (...msg: React.ReactNode[]): React.ReactNode => <ul key={`${code}${category}${msg}`}>{msg}</ul>,
          phoneNumber: <DteContactNumber value={contactNumber} />,
        }}
      />
      {showVerificationHistory && (
        <>
          <span className="dte-wismo-view-details-status-history-title">
            <FormattedMessage {...messages.status_history_title} />
          </span>
          <VerificationHistory verificationEvents={events} />
        </>
      )}
    </span>
  );
};

export default ModalContent;
