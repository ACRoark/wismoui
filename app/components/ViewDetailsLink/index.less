@import (reference) '../../styles/themes/index.less';

.@{dte-prefix}-view-details-modal-content {
  display: block;
  text-align: left;

  .@{dte-prefix}-view-details-status-history-title {
    color: @dte-gray-700;
    display: block;
    font-weight: @dte-wismo-font-weight-semi-bold;
    padding: 20px 0;
    text-align: left;
  }

  .website {
    font-weight: @dte-wismo-font-weight-semi-bold;
    text-decoration: underline;
  }
}
