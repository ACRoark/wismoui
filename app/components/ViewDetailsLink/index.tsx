import * as _ from 'lodash';
import React from 'react';
import { FormattedMessage } from 'react-intl';

import Button from 'components/Button';
import useModal from 'hooks/useModal';
import useThunkDispatch from 'hooks/useThunkDispatch';
import { trackButtonClick } from 'store/actions';
import IModalConfig from 'types/IModalConfig';
import IOrderRequest from 'types/IOrderRequest';
import getServiceDate from 'utils/getServiceDate';

import getServiceOrderEventsModal from './getServiceOrderEventsModal';
import getVerificationEventsModal from './getVerificationEventsModal';

import './index.less';
import messages from './messages';

interface IViewDetailsLinkProps {
  className?: string;
  orderCreatedDate: string;
  serviceRequest: IOrderRequest;
}

const ViewDetailsLink: React.FC<IViewDetailsLinkProps> = (props: IViewDetailsLinkProps): React.ReactElement => {
  const { className, orderCreatedDate, serviceRequest } = props;
  const { serviceOrderEvents, verificationEvents } = serviceRequest;

  let serviceOrderEventStatus;

  if (serviceOrderEvents && serviceOrderEvents.length) {
    if (_.last(serviceOrderEvents).latestStatus) {
      serviceOrderEventStatus = _.last(serviceOrderEvents).latestStatus.serviceOrderEventStatus;
    }
  }

  const dispatch = useThunkDispatch();
  const { showModal } = useModal();
  const serviceDate = getServiceDate(serviceRequest) || orderCreatedDate;
  const isVerificationComplete = serviceOrderEvents && serviceOrderEvents.length;

  const emptyModal = { content: null, title: null };

  const ServiceOrderEventsModal = serviceOrderEventStatus
    ? getServiceOrderEventsModal(serviceOrderEventStatus)
    : emptyModal;

  const VerificationEventsModal =
    verificationEvents && verificationEvents.length
      ? getVerificationEventsModal(verificationEvents, serviceDate)
      : emptyModal;

  const modal: IModalConfig = isVerificationComplete ? ServiceOrderEventsModal : VerificationEventsModal;

  const trackButtonAndOpenModal = (): void => {
    dispatch(trackButtonClick('view details link'));
    showModal(modal);
  };

  return (
    <Button className={className} onClick={trackButtonAndOpenModal} type="link">
      <FormattedMessage {...messages.title} />
    </Button>
  );
};

export default ViewDetailsLink;
