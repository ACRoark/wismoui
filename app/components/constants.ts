import IApplicationConfig from 'types/IApplicationConfig';

export const PageTitles = Object.freeze({
  ORDER_STATUS_PAGE: 'Track Your Order | DTE Energy',
  SEARCH_PAGE: 'Order Tracker | DTE Energy',
  SUMMARY_PAGE: 'Track Your Order | DTE Energy',
});

export const testConfig: IApplicationConfig = {
  flags: {
    canChangeLanguage: false,
    developerMode: false,
  },
  gigyaSiteKey: 'foo',
  googleTagManagerConfig: {
    auth: 'p3zyt1YCX84XB82sOrjMGw', // '',
    id: 'GTM-K2WP588', // 'GTM-PBB6QZ4',
    preview: 'env-1', // '',
  },
  loading: true,
  recaptchaSitekey: '6LeIxAcTAAAAAJcZVRqyHh71UMIEGNQ_MXjiZKhI',
  urls: {
    assistanceURL: 'https://www.dteenergy.com',
    backToDTE: 'https://www.dteenergy.com',
    backToMIMO: 'https://www.dteenergy.com',
    contactUs: 'https://www.contact.us.com',
    createAccount: 'https://www.create.account.com',
    decryptTokenApi: 'https://web-test.dteenergy.com/api/decryptToken?detokenizeValue=',
    documentSubmission: 'https://web-test.dteenergy.com/wps/wcm/connect/dte-web/quicklinks/web-form?detokenizeValue=',
    dynatraceScript:
      'https://dynatrace.dteenergy.com:9999/jstag/managed/98834406-1ae1-4257-992c-5b3c5f828584/968d8c7df07fe18c_complete.js',
    editPhoneNumber: 'https://www.edit.phone.number.com',
    editWantDate: 'https://www.edit.service.date.com',
    encryptTokenApi: 'https://web-test.dteenergy.com/api/encryptToken',
    findOrder: 'https://firebase.or.jsonserver/orders/{orderNumber}?name={name}',
    getOrder: 'https://firebase.or.jsonserver/orders/{orderNumber}',
    guestPay: 'https://www.guest.pay.com',
    headerLogo: 'https://www.header.logo.com',
    privacyPolicy: 'https://www.privacy.policy.com',
    searchOrders: 'https://firebase.or.jsonserver/orders',
    signIn: 'https://www.sign.in.com',
    termsAndConditions: 'https://www.terms.and.conditions.com',
  },
};
