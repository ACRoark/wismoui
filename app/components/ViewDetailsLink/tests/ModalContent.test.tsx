import React from 'react';

import { testConfig } from 'components/constants';
import useConfig from 'hooks/useConfig';
import { SUPPORTED_LOCALES } from 'locales';
import encryptionApiClient from 'services/encryptionApiClient';
import { createSnapshotWithIntl } from 'testing/utils';
import { IVerificationEvent } from 'types';

import ModalContent from '../ModalContent';

jest.mock('hooks/useConfig');
jest.mock('services/encryptionApiClient');

const mockConfig = useConfig as jest.MockedFunction<typeof useConfig>;
const mockEncryption = encryptionApiClient.encryptTokenAsync as jest.MockedFunction<typeof encryptionApiClient.encryptTokenAsync>;

const expectComponentToMatchSnapshot = (event: IVerificationEvent, events: IVerificationEvent[], language: string, orderCreatedDate: string): void => {
  const tree = createSnapshotWithIntl(<ModalContent event={event} events={events} orderCreatedDate={orderCreatedDate} />, language);

  expect(tree).toMatchSnapshot();
};

describe('ModalContent', (): void => {
  mockConfig.mockReturnValue({ ...testConfig });
  mockEncryption.mockReturnValue(new Promise((): string => ''));

  const languages = Object.keys(SUPPORTED_LOCALES);

  languages.forEach((language: string): void => {
    it('should render the modal content for AD', (): void => {
      const multipleBpemVerificationEvents: IVerificationEvent[] = [
        {
          verificationCategory: 'R975',
          verificationEventId: '123111',
          verificationEventStatusUpdates: [
            {
              createdAtDate: '20201005',
              createdAtTime: '091510',
              verificationCode: 'AD',
              verificationEventStatus: 'ACTIVE',
            },
          ],
        },
        {
          verificationCategory: 'R980',
          verificationEventId: '123112',
          verificationEventStatusUpdates: [
            {
              createdAtDate: '20201005',
              createdAtTime: '092010',
              verificationCode: 'PR',
              verificationEventStatus: 'ACTIVE',
            },
          ],
        },
      ];

      expectComponentToMatchSnapshot(multipleBpemVerificationEvents[0], multipleBpemVerificationEvents, language, '2020-04-19');
    });

    it('should render the modal content for AP', (): void => {
      const multipleBpemVerificationEvents: IVerificationEvent[] = [
        {
          verificationCategory: 'R975',
          verificationEventId: '123111',
          verificationEventStatusUpdates: [
            {
              createdAtDate: '20201005',
              createdAtTime: '091510',
              verificationCode: 'AP',
              verificationEventStatus: 'ACTIVE',
            },
          ],
        },
        {
          verificationCategory: 'R980',
          verificationEventId: '123112',
          verificationEventStatusUpdates: [
            {
              createdAtDate: '20201005',
              createdAtTime: '092010',
              verificationCode: 'PR',
              verificationEventStatus: 'ACTIVE',
            },
          ],
        },
      ];

      expectComponentToMatchSnapshot(multipleBpemVerificationEvents[0], multipleBpemVerificationEvents, language, '2020-04-19');
    });
  });
});
