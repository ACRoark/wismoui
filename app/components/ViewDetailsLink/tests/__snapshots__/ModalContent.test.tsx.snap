// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalContent should render the modal content for AD 1`] = `
<span
  className="dte-wismo-view-details-modal-content"
>
  <p>
    More information is needed to start service. Please visit our 
    <a
      className="website"
      href="https://web-test.dteenergy.com/wps/wcm/connect/dte-web/quicklinks/web-form?detokenizeValue="
      target="_blank"
    >
      Document Submission Portal
    </a>
     and select Residential Start Service. Next, fill out your information and then submit one of the following documents:
  </p>
  <ul>
    <li>
      Government-issued picture ID
    </li>
    <li>
      Social Security card
    </li>
    <li>
      Driver’s license
    </li>
    <li>
      Passport
    </li>
    <li>
      Native American tribal ID
    </li>
    <li>
      Military ID
    </li>
    <li>
      Detroit Municipal ID
    </li>
  </ul>
  <p>
    After your information is submitted, a Customer Service team member will contact you via the phone number provided within 2 business days.
  </p>
  <p>
    If you do not submit your information, DTE will close your order 30 days after 
    04/19/20
    .
  </p>
  <span
    className="dte-wismo-view-details-status-history-title"
  >
    Status History
  </span>
  <ul
    className="ant-timeline dte-wismo-timeline"
  >
    <li
      className="ant-timeline-item ant-timeline-item-last"
    >
      <div
        className="ant-timeline-item-tail"
      />
      <div
        className="ant-timeline-item-head ant-timeline-item-head-gray"
        style={
          Object {
            "borderColor": undefined,
          }
        }
      />
      <div
        className="ant-timeline-item-content"
      >
        Documents Received 
        10/05/20
      </div>
    </li>
  </ul>
</span>
`;

exports[`ModalContent should render the modal content for AD 2`] = `
<span
  className="dte-wismo-view-details-modal-content"
>
  <p>
    Se necesita más información para iniciar el servicio. Visite nuestro 
    <a
      className="website"
      href="https://web-test.dteenergy.com/wps/wcm/connect/dte-web/quicklinks/web-form?detokenizeValue="
      target="_blank"
    >
      Portal de envío de documentos
    </a>
     y seleccione Servicio de inicio residencial. Luego, complete su información y luego envíe uno de los siguientes documentos:
  </p>
  <ul>
    <li>
      Identificación con foto emitida por el gobierno
    </li>
    <li>
      Tarjeta de seguro Social
    </li>
    <li>
      Licencia de conducir
    </li>
    <li>
      Pasaporte
    </li>
    <li>
      Identificación tribal indígena americana
    </li>
    <li>
      ID militar
    </li>
    <li>
      ID municipal de Detroit
    </li>
  </ul>
  <p>
    Después de enviar su información, un miembro del equipo de Servicio al Cliente se comunicará con usted a través del número de teléfono proporcionado dentro de los 2 días hábiles.
  </p>
  <p>
    Si no envía su información, DTE cerrará su pedido 30 días después 
    19/04/20
    .
  </p>
  <span
    className="dte-wismo-view-details-status-history-title"
  >
    Historial de estado
  </span>
  <ul
    className="ant-timeline dte-wismo-timeline"
  >
    <li
      className="ant-timeline-item ant-timeline-item-last"
    >
      <div
        className="ant-timeline-item-tail"
      />
      <div
        className="ant-timeline-item-head ant-timeline-item-head-gray"
        style={
          Object {
            "borderColor": undefined,
          }
        }
      />
      <div
        className="ant-timeline-item-content"
      >
        Documentos Recibidos 
        05/10/20
      </div>
    </li>
  </ul>
</span>
`;

exports[`ModalContent should render the modal content for AP 1`] = `
<span
  className="dte-wismo-view-details-modal-content"
>
  In order to complete your start service order, please schedule the date/time 
  <a
    className="website"
    href="https://www.dteenergy.com"
    target="_blank"
  >
    online
  </a>
  .
</span>
`;

exports[`ModalContent should render the modal content for AP 2`] = `
<span
  className="dte-wismo-view-details-modal-content"
>
  Para completar su orden de inicio de servicio, programe la fecha / hora 
  <a
    className="website"
    href="https://www.dteenergy.com"
    target="_blank"
  >
    en línea
  </a>
  .
</span>
`;
